idaes/core/process_base.py

```diff
@@ -268,6 +268,9 @@ def report(self, time_point=0, dof=False, ostream=None, prefix=""):
                           f"Activated Blocks: {nb}")
 
         if performance is not None:
+            # PYLINT-WHY: pylint has no way of knowing that performance is supposed to be dict-like
+            # pylint: disable=unsubscriptable-object
+            # PYLINT-TODO: alternatively, have the function return an empty dict and test with `if performance:`
```

Comment:
I do not see a problem with this suggestion (although I cannot guarantee it won't break something somewhere else). I don't think there was a particular reason for having `get_performance_contents` return `None` by default.

Reply:
A quick check showed at least two unit tests in `test_process_base()` that would need to be updated. Apart from the (however minor) chance of breaking the existing API, I'm not sure I agree with pylint that this is an error: personally, for this and similar cases, I don't think there's anything particularly wrong with returning `None` as a sentinel value.

Seeing that all errors of the same type are related to the use of the same `_get_performance_contents()` in derived classes, I would propose to add this specific error (`E1128: Assigning result of a function call, where the function returns None (assignment-from-none)`) to the list of globally ignored checks, and leave the similar but semantically quite different `E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)` active, as IMO it covers a case that is much more likely to be accidental ("forgetting" to return a value where the client expects some, versus explicitly using `return None`).